File: Sorting and Searching/Binary Search.c
#include<stdio.h>

int BinarySearch(int arr[], int size, int element)
{
    int low = 0;
    int mid;
    int high = size -1;
    while (low <= high) 
    {
        mid = (low + high)/2;
        if (arr[mid] == element) 
        {
            return mid;
        }
        if (arr[mid] < element) 
        {
            low = mid +1;
        }
        else 
        {
            high = mid -1;
        }
    }
    return -1;
    
}

int main()
{
    int arr[] = {11, 22, 33, 44, 55, 66, 77, 88, 99, 100};
    int size = sizeof(arr)/sizeof(int);
    int element = 11;
    int searchIndex = BinarySearch(arr, size, element);
    printf("The element %d was found at index %d\n", element, searchIndex);
    return 0;
}

File: Algorithms/Recursion/Binary_Search.java
public class Binary_Search {
    static boolean Search(int[] arr, int s, int e, int key) {
        if (s > e)
            return false;
        int mid = (s + e) / 2;
        if (arr[mid] == key)
            return true;
        else if (arr[mid] > key)
            return Search(arr, s, mid - 1, key);
        else
            return Search(arr, mid + 1, e, key);
    }

    public static void main(String[] a) {
        int[] arr = { 2, 4, 6, 8, 17, 19 };
        int n = arr.length;
        int key = 22;
        System.out.print(Search(arr, 0, n - 1, key));
    }
}